docsrc/content/basics/internationalization.go

```diff
@@ -0,0 +1,97 @@
+package basics
+
+import (
+	"path"
+
+	"github.com/qor5/docs/docsrc/examples/example_basics"
+	"github.com/qor5/docs/docsrc/generated"
+	"github.com/qor5/docs/docsrc/utils"
+	"github.com/theplant/docgo"
+	"github.com/theplant/docgo/ch"
+)
+
+var I18n = docgo.Doc(
+	docgo.Markdown(`
+The [i18n package](https://github.com/qor5/x/tree/master/i18n) provides support for internationalization (i18n) in Go applications. 
+It allows you to define and load messages in multiple languages, and provides a mechanism 
+for selecting the appropriate language for a given user request.
+    `),
+	docgo.Markdown(`
+## Getting Started
+To use the i18n package, you first need to import it into your Go application:
+`),
+	ch.Code(`import "github.com/qor5/x/i18n"`).Language("go"),
+	docgo.Markdown(`
+Next, create a new ~Builder~ instance using the ~New()~ function. 
+If you want to use it in QOR5, use the ~I18n()~ on ~presets.Builder~:
+`),
+	ch.Code(generated.I18nNew).Language("go"),
+	docgo.Markdown(`
+The ~Builder~ struct is the central point of the i18n package. 
+It holds the supported languages, the messages for each module in each language, 
+and the configuration for retrieving the language preference.
+`),
+	docgo.Markdown(`
+## Supporting Languages
+To support multiple languages in your web application, you need to define the languages that you support. 
+You can do this by calling the ~SupportLanguages~ function on the ~Builder~ struct:
+`),
+	ch.Code(generated.I18nSupportLanguages).Language("go"),
+	docgo.Markdown(`
+By default, the ~Builder~ struct only supports the English language. 
+You can add more languages by calling the ~SupportLanguages~ function and passing in the language tags.
+`),
+	docgo.Markdown(`
+## Registering Module Messages
+Once you have defined the languages that you support, you need to register messages for each module in each language. 
+You can do this by calling the ~RegisterForModule~ function on the ~Builder~ struct:
+`),
+	ch.Code(generated.I18nRegisterForModule).Language("go"),
+	docgo.Markdown(`
+The ~RegisterForModule~ function takes three arguments: the language tag, the module name, 
+and a pointer to a struct that implements the Messages interface. 
+The Messages interface is an empty interface that you can use to define your own messages.
+
+Such a struct might look like this:
+`),
+	ch.Code(generated.I18nMessagesExample).Language("go"),
```

Comment:
@moluuser TODO: clarify how and where users can find those preset keys in the system.